src/__tests__/main.test.ts

```diff
@@ -339,7 +342,121 @@ test("upsert where clause", async (t) => {
     t.is(res.data.allRoles.edges[0].node.projectName, "sales");
     t.is(res.data.allRoles.edges[0].node.title, "director");
     t.is(res.data.allRoles.edges[0].node.name, "frank");
-    t.is(res.data.allRoles.edges[0].node.rank, 2);
+    t.is(res.data.allRoles.edges[0].node.rank, 1);
+
+    // assert only one record
+    t.is(res.data.allRoles.edges.length, 1);
+  }
+});
+
+test("upsert where clause omit onConflictUpdate", async (t) => {
+  const upsertDirector = async ({
+    projectName = "sales",
+    title = "director",
+    name = "jerry",
+    rank = 1,
+  }: {
+    projectName?: string;
+    title?: string;
+    name?: string;
+    rank?: number;
+  }) => {
+    const query = nanographql(`
+      mutation {
+        upsertRole(onConflict: {doUpdate: {name: ignore, updated: current_timestamp}}, where: {
+          projectName: "sales",
+          title: "director"
+        },
+        input: {
+          role: {
+            projectName: "${projectName}",
+            title: "${title}",
+            name: "${name}",
+            rank: ${rank}
+          }
+        }) {
+          clientMutationId
+        }
+      }
+    `);
+    return execGqlOp(t, query);
+  };
+  {
+    // add director
+    await upsertDirector({ name: "jerry" });
+    const res = await fetchAllRoles(t);
+    t.is(res.data.allRoles.edges.length, 1);
+    t.is(res.data.allRoles.edges[0].node.projectName, "sales");
+    t.is(res.data.allRoles.edges[0].node.title, "director");
+    t.is(res.data.allRoles.edges[0].node.name, "jerry");
+    t.is(res.data.allRoles.edges[0].node.updated, null);
```

Comment:
will you experiment using [t.like](https://github.com/avajs/ava/blob/main/docs/03-assertions.md#likeactual-selector-message) here, so we can reduce the large qty of field lookups? not that what you've done is a problem, but i think it could be slick :)